### PR TITLE
feat: onboard Yandex Metrica Offline Events Destination

### PR DESCRIPTION
**Pull request by yashasvibajpai**
## What are the changes introduced in this PR?

Onboard Yandex Metrica Offline Events Destination

## What is the related Linear task?

Resolves INT-1838

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3232/test-report.html

File: src/cdk/v2/destinations/yandex_metrica_offline_events/config.js
Comment (ItsSudip) [inline src/cdk/v2/destinations/yandex_metrica_offline_events/config.js:1]: Where are we using this export?

File: src/cdk/v2/destinations/yandex_metrica_offline_events/procWorkflow.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    path: ../../../../v0/util
+  - path: ./utils
+
+steps:
+  - name: validateInput
+    template: |
+      let messageType = .message.type;
+      $.assert(messageType, "message Type is not present. Aborting message.");
+      $.assert(.message.type.toLowerCase() ==='identify' || .message.type.toLowerCase() ==='track', "Event type " + .message.type.toLowerCase() + " is not supported. Aborting message.");
+      $.assert(.message.traits || .message.properties, "Message traits/properties not present. Aborting message.");
```
Comment (ItsSudip) [inline src/cdk/v2/destinations/yandex_metrica_offline_events/procWorkflow.yaml:19]: We will get identify calls only why are we validating for track?

File: src/cdk/v2/destinations/yandex_metrica_offline_events/procWorkflow.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      let identifierValue = .message.context.externalId[0].id;
+      data = $.setIdentifier(data, identifierType, identifierValue)
+      data
+
+  - name: buildResponseForProcessTransformation
+    description: build response
+    template: |
+      const response = $.defaultRequestConfig();
+      response.body.JSON = $.outputs.prepareData
+      console.log(JSON.stringify(response));
```
Comment (ItsSudip) [inline src/cdk/v2/destinations/yandex_metrica_offline_events/procWorkflow.yaml:34]:
```suggestion
```

File: src/cdk/v2/destinations/yandex_metrica_offline_events/procWorkflow.yaml (hunk 1 above)
Comment (yashasvibajpai) [reply src/cdk/v2/destinations/yandex_metrica_offline_events/procWorkflow.yaml]: Removed, thanks

File: src/cdk/v2/destinations/yandex_metrica_offline_events/config.js
Comment (yashasvibajpai) [reply src/cdk/v2/destinations/yandex_metrica_offline_events/config.js:1]: I referenced this change from Eloqua. Should I remove it?

Comment (ItsSudip) [reply src/cdk/v2/destinations/yandex_metrica_offline_events/config.js:1]: If it is not required we should remove this

Comment (yashasvibajpai) [reply src/cdk/v2/destinations/yandex_metrica_offline_events/config.js:1]: error": "Binding not found: ./config:
The execution fails with this error, @koladilip should this remain then?

File: test/integrations/destinations/yandex_metrica_offline_events/processor/data.ts [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+              workspaceId: 'wspId',
+            },
+          },
+        ],
+      },
+    },
+  },
+  {
+    name: 'yandex_metrica_offline_events',
+    description: 'Successful identify event with UserId identifier type',
```
Comment (sanpj2292) [inline test/integrations/destinations/yandex_metrica_offline_events/processor/data.ts:298]: This test-case has wrong description. Can we correct it ?

File: src/cdk/v2/destinations/yandex_metrica_offline_events/procWorkflow.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      let messageType = .message.type;
+      $.assert(messageType, "message Type is not present. Aborting message.");
+      $.assert(.message.type.toLowerCase() ==='identify', "Event type " + .message.type.toLowerCase() + " is not supported. Aborting message.");
+      $.assert(.message.traits || .message.properties, "Message traits/properties not present. Aborting message.");
+
+  - name: prepareData
+    template: |
+      let data = .message.traits
+      data = $.validateData(data)
+      let identifierType = .message.context.externalId[0].identifierType;
```
Comment (sanpj2292) [inline src/cdk/v2/destinations/yandex_metrica_offline_events/procWorkflow.yaml:25]: From the source-code, it seems like `externalId` is an absolute need. Will we have it for all or should we have a validation for `externalId` as well ?

File: test/integrations/destinations/yandex_metrica_offline_events/processor/data.ts [hunk 5]
```diff
@@ -0,0 +1,384 @@
+export const data = [
```
Comment (sanpj2292) [inline test/integrations/destinations/yandex_metrica_offline_events/processor/data.ts:1]: Can we add test-cases for missing lines ?

File: test/integrations/destinations/yandex_metrica_offline_events/processor/data.ts (hunk 3 above)
Comment (yashasvibajpai) [reply test/integrations/destinations/yandex_metrica_offline_events/processor/data.ts]: Apologies, Fixed it.

File: src/cdk/v2/destinations/yandex_metrica_offline_events/procWorkflow.yaml (hunk 4 above)
Comment (yashasvibajpai) [reply src/cdk/v2/destinations/yandex_metrica_offline_events/procWorkflow.yaml:25]: We will have it from sources, but have added the test case based on existing validation

File: test/integrations/destinations/yandex_metrica_offline_events/processor/data.ts (hunk 5 above)
Comment (yashasvibajpai) [reply test/integrations/destinations/yandex_metrica_offline_events/processor/data.ts:1]: Added.

**Review by sanpj2292 — APPROVED** [review]

**Review by sanpj2292 — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**